### PR TITLE
Python 3 support

### PR DESCRIPTION
**Pull request by hombit**
This PR adds Python 3 support using [`six` module](https://github.com/benjaminp/six). Also, it requires to switch from `oauth` to its fork [`oauth2`](https://github.com/joestump/python-oauth2). `oauth2` supports Python 3 but don't compatible with original oauth, so I fix `auth.py` to support it.

This code hasn't been tested enough, I've checked that both samples work on both Python 2 and 3 and checked photos upload. It shouldn't be upload on pypi as a new version of the package as is. However, I ask you to accept this PR and ask community to test it. We can write some unit tests as a next step.

### CODE REVIEW

**Comment by Flimm** [comment]
Hello python-flickr-api maintainers. Is there any hope of this pull request getting accepted? Python 3 support would be nice.

File: flickr_api/auth.py [hunk 1]
```diff
@@ -75,33 +71,33 @@ def __init__(self, key=None, secret=None, callback=None,
...
+            req = oauth2.Request(method="GET",
+                                 url=TOKEN_REQUEST_URL,
+                                 parameters=params)
+            req.sign_request(oauth2.SignatureMethod_HMAC_SHA1(),
                              self.consumer, None)
-            resp = urllib2.urlopen(req.to_url())
-            request_token = dict(urlparse.parse_qsl(resp.read()))
-            self.request_token = oauth.OAuthToken(
+            resp = urllib.request.urlopen(req.to_url())
+            request_token = dict(urllib.parse.parse_qsl(resp.read()))
```
Comment (tomquirk) [inline flickr_api/auth.py]: This is returning byte-encoded strings and causes the lookup in the next line to fail.

I tried messing with the parse_qsl params but couldnt get anywhere, but this (albeit dodgy) worked:
```python
request_token = {key.decode(): value.decode() for key, value in urllib.parse.parse_qsl(resp.read())}
```

Comment (alexis-mignon) [reply flickr_api/auth.py]: Should I merge the PR and let you correct this ?

Comment (tomquirk) [reply flickr_api/auth.py]: Yea sounds good, I'll merge it in. I've been messing with everything for a bit now and everything seems fine. I'll swap in my fix ASAP

**Pull request merged**